### PR TITLE
explanatory error message on delete attempt of default system offering

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #8882 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@DaanHoogland I have changed the base to 4.18 according to the milestone.

**Review by vishesh92 — APPROVED** [review]
clgtm

**Comment by vishesh92** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -4204,7 +4204,7 @@ public boolean deleteServiceOffering(final DeleteServiceOfferingCmd cmd) {
         }
 
         if (offering.getDefaultUse()) {
-            throw new InvalidParameterValueException("Default service offerings cannot be deleted");
+            throw new InvalidParameterValueException(String.format("The system service offerings [%s], is marked for default use and cannot be deleted", offering.getDisplayText()));
```
Comment (gpordeus) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:4207]:
```suggestion
            throw new InvalidParameterValueException(String.format("The system service offering [%s] is marked for default use and cannot be deleted", offering.getDisplayText()));
```

**Review by gpordeus — APPROVED** [review]

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by vishesh92 — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**